ludwig/visualize.py

```diff
@@ -3618,7 +3618,7 @@ def frequency_vs_f1(
 
             # get np arrays of frequencies, f1s and labels
             idx2freq = {metadata[of_name]["str2idx"][key]: val for key, val in metadata[of_name]["str2freq"].items()}
-            freq_np = np.array([idx2freq[class_id] for class_id in sorted(idx2freq)], dtype=np.int32)
+            freq_np = np.array([idx2freq[class_id] for class_id in sorted(idx2freq)][:k], dtype=np.int32)
```

Comment:
I find it weird that in line 3608 there is:
```
if k > 0:
     class_names = class_names[:k]
```

This implies to me that if k == 0 we should be showing all classes.... like k==0 means don't truncate.  If so, then maybe we should change that to:
```
k = top_n_classes[i]
if k <= 0:
    k = len(class_names)
class_names = class_names[:k]
```

Reply:
@dantreiman Afaik, `top_n_classes` is a list with a single item, which is just the number of top classes. I don't know why it was written this way but I noticed this in many of the other visualization tests, so I think this may not work, @hungcs you may want to double check to be sure

Reply:
So maybe something like:

``` 
if k <= len(class_names):
    class_names = class_names[:k]
```
?

Reply:
But then yeah, I agree that if k == 0, then this doesn't quite work because it won't produce anything.

Reply:
I'm just assuming that `if k > 0:` was put in there for a good reason :-).

We could also do
```
if k > 0:
    freq_np = freq_np[:k]
```

up to you @hungcs, and feel free to ignore the part where I wrote top_n_classes[i].